tests/sqlparser_common.rs

```diff
@@ -2744,6 +2747,21 @@ fn parse_create_sources_with_like_regex() {
     }
 }
 
+#[test]
+fn parse_show_objects_with_like_regex() {
+    let sql = "SHOW TABLES LIKE '%foo%'";
+    match verified_stmt(sql) {
+        Statement::ShowObjects { object_type, like } => {
+            match like {
+                Some(value) => assert_eq!("%foo%", value),
+                None => unimplemented!(),
+            }
+            assert_eq!(ObjectType::Table, object_type);
+        }
+        _ => assert!(false),
```

Comment:
I generally prefer `panic!("invalid show objects")` because the error messages are better, but it's not important